src/main/java/com/google/devtools/build/lib/syntax/SkylarkList.java

```diff
@@ -521,15 +522,15 @@ public void set(int index, E value, Location loc, Mutability mutability) throws
       name = "extend",
       doc = "Adds all items to the end of the list.",
       parameters = {
-          @Param(name = "items", type = SkylarkList.class, doc = "Items to add at the end.")
+          @Param(name = "items", type = Object.class, doc = "Items to add at the end.")
       },
       useLocation = true,
       useEnvironment = true
     )
     public Runtime.NoneType extend(
-        SkylarkList<E> items, Location loc, Environment env)
+        Object items, Location loc, Environment env)
         throws EvalException {
-      addAll(items, loc, env.mutability());
+      addAll(new ArrayList(EvalUtils.toCollection(items, loc, env)), loc, env.mutability());
```

Comment:
This incurs an unnecessary copy that was not present in the old code. Essentially the problem has been pushed to 'new ArrayList', which has a raw type, which ought to be a compile error if our build was correctly configured. Let's revert to the previous version.

(I think the right solution is to abolish the type parameter in StarlarkList, so that it extends `Collection<Object>`. If you pass a `StarlarkList<X>` to the interpreter, you have no guarantee that it doesn't insert objects of class Y into the list. The only reason this doesn't happen today is that Blaze never passes mutable lists (other than of `<Object>)` to Starlark functions, but an application that did would start to see very confusing crashes. 
But that's a much bigger change.)

Reply:
fyi, I've merged this PR using Friday's state (without the unnecessary copy, but with the cast).